src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.ts

```diff
@@ -0,0 +1,107 @@
+import { TServerEntityId } from 'src/app/common/http/common.http.types';
+import { checkInputError } from 'src/app/common/utils/forms/common-utils-forms-form-utils';
+import { WalletTransactionType } from 'src/app/domains/transactions/domains.transactions.constants';
+import { WalletsDetailsTransaction } from '../pages-wallet-details-item.model';
+import { IWalletTransactionModalFormType } from './pages-wallet-transaction.editor.types';
+
+export const regexAmount = /^\d+(\.\d{1,2})?$/;
+
+@Component({
+  selector: 'app-transaction-editor',
+  templateUrl: './pages-wallet-transaction-editor.component.html',
+  styleUrls: [ './pages-wallet-transaction-editor.component.scss' ],
+})
+export class PagesWalletTransactionEditorComponent {
+  public selectTransactionsTypes: Record<string, WalletTransactionType> = {
+    'Income': WalletTransactionType.Incomes,
+    'Expense': WalletTransactionType.Expenses,
+  };
+
+  public form!: FormGroup<IWalletTransactionModalFormType>;
+
+  private transactionId: TServerEntityId | null = null;
+
+  private walletId: TServerEntityId;
+
+  public constructor(
+    private readonly dialogRef: MatDialogRef<PagesWalletTransactionEditorComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: WalletsDetailsTransaction,
+  ) {
+    this.transactionId = this.data.id;
+    this.walletId = this.data.walletId;
+
+    this.form = new FormGroup<IWalletTransactionModalFormType>({
+      amount: new FormControl(this.data.amount === 0 ? null : this.data.amount, {
+        validators: [
+          Validators.required,
+          Validators.pattern(regexAmount),
+          this.amountBiggerThanZero(),
+        ],
+        nonNullable: false,
+      }),
+      description: new FormControl(this.data.description),
+      date: new FormControl(this.data.transactionDate, {
+        validators: [
+          Validators.required,
+        ],
+        nonNullable: false,
+      }),
+      type: new FormControl(this.data.type, {
+        validators: [
+          Validators.required,
+        ],
+        nonNullable: true,
+      }),
+    });
+  }
+
+  public get amountIsNotProvided(): boolean {
+    return checkInputError(this.form, 'amount', 'required');
+  }
+
+  public get amountIsZero(): boolean {
+    return checkInputError(this.form, 'amount', 'amountIsZero');
+  }
+
+  public get amountHasInvalidFormat(): boolean {
+    return checkInputError(this.form, 'amount', 'pattern');
+  }
+
+  public get dateIsNotProvided(): boolean {
+    return checkInputError(this.form, 'date', 'required');
+  }
+
+  public save(): void {
+    if (this.form.invalid) {
+      return;
+    }
+
+    this.dialogRef.close(WalletsDetailsTransaction.create({
+      id: this.transactionId ?? undefined,
+      amount: parseFloat(this.form.controls.amount.value!.toString()),
+      creationDate: this.form.controls.date.value!.toString(),
+      type: this.form.controls.type.value!,
+      description: this.form.controls.description.value || null,
```

Comment:
No luck with this one, both solutions sounds shitty to me:
- adding eslint comment to ignore this one (because we will have to add it everytime in such scenario)
- disabling this rule (because we actually should use `??` whenever we can)
So I think you can go with if / ternary and I will think about some long term global solution.